src/components/views/rooms/MessageComposer.tsx

```diff
@@ -286,12 +286,16 @@ export default class MessageComposer extends React.Component<IProps, IState> {
             showStickers: false,
             showStickersButton: SettingsStore.getValue("MessageComposerInput.showStickersButton"),
             showPollsButton: SettingsStore.getValue("feature_polls"),
-            showLocationButton: SettingsStore.getValue("feature_location_share"),
+            showLocationButton: (
+                SettingsStore.getValue("feature_location_share") &&
+                SettingsStore.getValue("MessageComposerInput.showLocationButton")
+            ),
```

Comment:
This should've leveraged Settings controllers to be always off when `feature_location_share` is off, e.g

https://github.com/matrix-org/matrix-react-sdk/blob/7b95202f44ebb3136adb96798145aab424c26f1f/src/settings/Settings.tsx#L857-L865